client/fs_endpoint_test.go

```diff
@@ -215,48 +198,31 @@ func TestFS_List_NoAlloc(t *testing.T) {
 }
 
 func TestFS_List(t *testing.T) {
-	t.Skip("missing mock driver plugin implementation")
 	t.Parallel()
 	require := require.New(t)
 
-	// Start a client
-	c, cleanup := TestClient(t, nil)
+	// Start a server and client
+	s := nomad.TestServer(t, nil)
+	defer s.Shutdown()
+	testutil.WaitForLeader(t, s.RPC)
+
+	c, cleanup := TestClient(t, func(c *config.Config) {
+		c.Servers = []string{s.GetConfig().RPCAddr.String()}
+	})
 	defer cleanup()
 
 	// Create and add an alloc
-	a := mock.Alloc()
-	task := a.Job.TaskGroups[0].Tasks[0]
-	task.Driver = "mock_driver"
-	task.Config = map[string]interface{}{
+	job := mock.BatchJob()
+	job.TaskGroups[0].Count = 1
+	job.TaskGroups[0].Tasks[0].Config = map[string]interface{}{
 		"run_for": "500ms",
 	}
-	c.addAlloc(a, "")
-
-	// Wait for the client to start it
-	testutil.WaitForResult(func() (bool, error) {
-		ar, ok := c.allocs[a.ID]
-		if !ok {
-			return false, fmt.Errorf("alloc doesn't exist")
-		}
-
-		alloc := ar.Alloc()
-		running := false
-		for _, s := range alloc.TaskStates {
-			if s.State == structs.TaskStateRunning {
-				running = true
-			} else {
-				running = false
-			}
-		}
-
-		return running, fmt.Errorf("tasks not running")
-	}, func(err error) {
-		t.Fatal(err)
-	})
+	// Wait for alloc to be running
+	alloc := testutil.WaitForRunning(t, s.RPC, job)[0]
```

Comment:
Same issue as above (I've probably made this mistake too... it's seems like a very unlikely race).